src/OpenLoco/src/GameCommands/VehicleRefit.cpp

```diff
@@ -0,0 +1,102 @@
+#include "Economy/Expenditures.h"
+#include "Entities/EntityManager.h"
+#include "GameCommands.h"
+#include "Objects/CargoObject.h"
+#include "Objects/ObjectManager.h"
+#include "Objects/VehicleObject.h"
+#include "Types.hpp"
+#include "Ui/WindowManager.h"
+#include "Vehicles/Vehicle.h"
+#include <OpenLoco/Interop/Interop.hpp>
+#include <OpenLoco/Utility/Numeric.hpp>
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::GameCommands
+{
+    // 0x0042F6DB
+    static uint32_t vehicleRefit(const VehicleRefitArgs& args, const uint8_t flags)
+    {
+        setExpenditureType(ExpenditureType::TrainRunningCosts);
+
+        try
+        {
+            Vehicles::Vehicle train(args.head);
+            auto& head = train.head;
+
+            if (!head->canBeModified())
+            {
+                return FAILURE;
+            }
+
+            if (!sub_431E6A(head->owner))
+            {
+                return FAILURE;
+            }
+
+            if (train.cars.empty())
+            {
+                setErrorText(StringIds::empty);
+                return 0;
+            }
+
+            auto car = train.cars.firstCar;
+            auto* vehObj = ObjectManager::get<VehicleObject>(car.body->objectId);
+
+            if (!vehObj->hasFlags(VehicleObjectFlags::refittable))
+            {
+                setErrorText(StringIds::empty);
+                return 0;
+            }
+
+            auto carIter = train.cars.begin();
+            carIter++;
```

Comment:
This looks wrong. I'm not sure what you are doing here. Shouldn't it just be `car.body` from L44

Reply:
Nice catch. Much cleaner this way, and works just as well.